Sources/Sentry/SentryFramesTracker.m

```diff
@@ -29,6 +30,7 @@
 
 @property (nonatomic, strong, readonly) SentryDisplayLinkWrapper *displayLinkWrapper;
 @property (nonatomic, assign) CFTimeInterval previousFrameTimestamp;
+@property (nonatomic) uint64_t previousFrameSystemTimestamp;
```

Comment:
Added in https://github.com/getsentry/sentry-cocoa/pull/2545/commits/325c1a1e1b502272b9e902874382849990c065a1. We previously used a bad compiler directive in the tests, so some assertions we wanted to use weren't actually being compiled. That is fixed and I added a test point for how we gather frame rates as well, for profiling.

I don't think it would be worth the effort to mock the system clock to validate the actual timestamp information for these; this is basically done implicitly at an integration level in `SentryProfilerSwiftTests`, where we ensure we've extracted the expected amount of slow/frozen frames and frame rates recorded during a transaction. 

Let me know if you had anything else in mind by way of testing this.